worker/worker.go

```diff
@@ -0,0 +1,149 @@
+package worker
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	"os/exec"
+	"sync"
+	"syscall"
+)
+
+const (
+	JSTATUS_START   = "running"
+	JSTATUS_END     = "finished"
+	JSTATUS_STOPPED = "stopped"
+)
+
+//Worker defines the operations to manage Jobs.
+type Worker interface {
+	Start(cmdName string, args []string) (string, error)
+	Stop(jobID string) error
+	GetStatus(jobID string) (Status, error)
+	GetOutput(ctx context.Context, jobID string, userID string) (chan string, error)
```

Comment:
```suggestion
	GetOutput(ctx context.Context, jobID string, userID string) (<-chan string, error)
```

Consider making this a read only channel at the API level, so it's clear to callers that something else is doing the writing.